mono_repo/lib/src/pubspec.dart

```diff
@@ -43,12 +43,35 @@ abstract class DependencyData {
         return new GitData.fromData(git);
       }
 
+      final sdk = mapData['sdk'];
+      if (sdk != null) {
+        return new SdkData(sdk);
+      }
+
       throw new ArgumentError.value(
           data, 'data', 'No clue how to deal with `$data`.');
     }
   }
 }
 
+class SdkData implements DependencyData {
+  final String name;
+
+  @override
+  DependencyType get type => DependencyType.sdk;
+
+  factory SdkData(Object data) {
+    if (data is String) {
+      return new SdkData._(name: data);
+    } else {
+      throw new ArgumentError.value(
+          data, 'data', 'Does not support provided value.');
+    }
+  }
+
+  SdkData._({this.name});
```

Comment:
Would we ever expect `name` to be null or not provided? If not, make it positional.